Regression: duplicate 'fixt' when overriding parametrized fixture with parametrize

This test:

```python
import pytest

@pytest.fixture(params=[1, 2])
def fixt(request):
    return request.param

@pytest.mark.parametrize('fixt, val', [(3, 'x'), (4, 'x')])
def test_foo(fixt, val):
    pass
```

works with 2.7.2 but fails since #926 / 41cef6f5f2a20b918bb9b9204ca6c91b47442d15:

```
.venv/lib/python3.4/site-packages/pluggy.py:724: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
.venv/lib/python3.4/site-packages/pluggy.py:338: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.venv/lib/python3.4/site-packages/pluggy.py:333: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
.venv/lib/python3.4/site-packages/pluggy.py:595: in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
.venv/lib/python3.4/site-packages/pluggy.py:249: in _wrapped_call
    wrap_controller.send(call_outcome)
_pytest/python.py:285: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
_pytest/python.py:442: in _genfunctions
    self.ihook.pytest_generate_tests(metafunc=metafunc)
.venv/lib/python3.4/site-packages/pluggy.py:724: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
.venv/lib/python3.4/site-packages/pluggy.py:338: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.venv/lib/python3.4/site-packages/pluggy.py:333: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
.venv/lib/python3.4/site-packages/pluggy.py:596: in execute
    res = hook_impl.function(*args)
_pytest/python.py:195: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
_pytest/python.py:967: in parametrize
    param_index)
_pytest/python.py:819: in setmulti
    self._checkargnotcontained(arg)
_pytest/python.py:802: in _checkargnotcontained
    raise ValueError("duplicate %r" %(arg,))
E   ValueError: duplicate 'fixt'
```

It only seems to fail if:

* The fixture is parametrized
* It's overridden by a parametrized argument
* There is at least one other parametrized argument.

I agree overriding a fixture this way is a bit weird, but I think it should still work.

/cc @untitaker @bubenkoff 

by the way - thanks to git bisect this was easy to find out:

```
git bisect start
git bisect bad master
git bisect good 2.7.2
git bisect run ./.venv/bin/py.test test_foo.py
```

no:

```
$ mkdir foo
$ cd foo
$ ls
$ virtualenv .venv
Running virtualenv with interpreter /usr/bin/python2
New python executable in .venv/bin/python2
Also creating executable in .venv/bin/python
Installing setuptools, pip...done.
$ ./.venv/bin/pip install git+https://github.com/pytest-dev/pytest.git
Downloading/unpacking git+https://github.com/pytest-dev/pytest.git
  Cloning https://github.com/pytest-dev/pytest.git to /tmp/pip-zpWML1-build
  Running setup.py (path:/tmp/pip-zpWML1-build/setup.py) egg_info for package from git+https://github.com/pytest-dev/pytest.git
    
    Installed /tmp/pip-zpWML1-build/setuptools_scm-1.7.0-py2.7.egg
    your setuptools is too old (<12)
    setuptools_scm functionality is degraded
    
Downloading/unpacking py>=1.4.29 (from pytest==2.7.3.dev380-ng553aef5)
  Downloading py-1.4.30-py2.py3-none-any.whl (81kB): 81kB downloaded
Downloading/unpacking pluggy>=0.3.0,<0.4.0 (from pytest==2.7.3.dev380-ng553aef5)
  Downloading pluggy-0.3.0-py2.py3-none-any.whl
Installing collected packages: py, pluggy, pytest
  Running setup.py install for pytest
    your setuptools is too old (<12)
    setuptools_scm functionality is degraded
    
    Installing py.test-2.7 script to /home/florian/foo/.venv/bin
    Installing py.test script to /home/florian/foo/.venv/bin
Successfully installed py pluggy pytest
Cleaning up...
$ cat > test_foo.py
$ ./.venv/bin/py.test test_foo.py 
========================================== test session starts ==========================================
platform linux2 -- Python 2.7.9, pytest-2.7.3.dev380+ng553aef5, py-1.4.30, pluggy-0.3.0
rootdir: /home/florian/foo, inifile: 
collected 0 items / 1 errors 

================================================ ERRORS =================================================
_____________________________________ ERROR collecting test_foo.py ______________________________________
.venv/local/lib/python2.7/site-packages/pluggy.py:724: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
.venv/local/lib/python2.7/site-packages/pluggy.py:338: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.venv/local/lib/python2.7/site-packages/pluggy.py:333: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
.venv/local/lib/python2.7/site-packages/pluggy.py:595: in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
.venv/local/lib/python2.7/site-packages/pluggy.py:249: in _wrapped_call
    wrap_controller.send(call_outcome)
.venv/local/lib/python2.7/site-packages/_pytest/python.py:285: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
.venv/local/lib/python2.7/site-packages/_pytest/python.py:442: in _genfunctions
    self.ihook.pytest_generate_tests(metafunc=metafunc)
.venv/local/lib/python2.7/site-packages/pluggy.py:724: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
.venv/local/lib/python2.7/site-packages/pluggy.py:338: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.venv/local/lib/python2.7/site-packages/pluggy.py:333: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
.venv/local/lib/python2.7/site-packages/pluggy.py:596: in execute
    res = hook_impl.function(*args)
.venv/local/lib/python2.7/site-packages/_pytest/python.py:195: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
.venv/local/lib/python2.7/site-packages/_pytest/python.py:967: in parametrize
    param_index)
.venv/local/lib/python2.7/site-packages/_pytest/python.py:819: in setmulti
    self._checkargnotcontained(arg)
.venv/local/lib/python2.7/site-packages/_pytest/python.py:802: in _checkargnotcontained
    raise ValueError("duplicate %r" %(arg,))
E   ValueError: duplicate 'fixt'
======================================== 1 error in 0.22 seconds ========================================

--- a/_pytest/python.py
+++ b/_pytest/python.py
@@ -1830,7 +1830,10 @@ class FixtureManager:
                 if fixturedef.params is not None:
                     func_params = getattr(getattr(metafunc.function, 'parametrize', None), 'args', [[None]])
                     # skip directly parametrized arguments
-                    if argname not in func_params:
+                    argnames = func_params[0]
+                    if not isinstance(argnames, (tuple, list)):
+                        argnames = [x.strip() for x in argnames.split(",") if x.strip()]
+                    if argname not in func_params and argname not in argnames:
                         metafunc.parametrize(argname, fixturedef.params,
                                              indirect=True, scope=fixturedef.scope,
                                              ids=fixturedef.ids)

--- a/testing/python/fixture.py
+++ b/testing/python/fixture.py
@@ -1616,6 +1616,10 @@ class TestFixtureMarker:
 
     @pytest.mark.parametrize('param_args', ["'fixt, val'", "'fixt,val'", "['fixt', 'val']", "('fixt', 'val')"])
     def test_override_parametrized_fixture_issue_979(self, testdir, param_args):
+        """Make sure a parametrized argument can override a parametrized fixture.
+
+        This was a regression introduced in the fix for #736.
+        """
         testdir.makepyfile("""
             import pytest
 

--- a/testing/python/fixture.py
+++ b/testing/python/fixture.py
@@ -1614,6 +1614,22 @@ class TestFixtureMarker:
         reprec = testdir.inline_run()
         reprec.assertoutcome(passed=9)
 
+    @pytest.mark.parametrize('param_args', ["'fixt, val'", "'fixt,val'", "['fixt', 'val']", "('fixt', 'val')"])
+    def test_override_parametrized_fixture_issue_979(self, testdir, param_args):
+        testdir.makepyfile("""
+            import pytest
+
+            @pytest.fixture(params=[1, 2])
+            def fixt(request):
+                return request.param
+
+            @pytest.mark.parametrize(%s, [(3, 'x'), (4, 'x')])
+            def test_foo(fixt, val):
+                pass
+        """ % param_args)
+        reprec = testdir.inline_run()
+        reprec.assertoutcome(passed=2)
+
     def test_scope_session(self, testdir):
         testdir.makepyfile("""
             import pytest

